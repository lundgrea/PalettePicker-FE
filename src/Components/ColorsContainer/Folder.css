.Folder {
  display: flex;
  font-size: 16px;
}

.folder_delete {
  height: 25px;
  margin: 10px;
}
.folder_delete:hover {
  filter: invert(0.5)
}

.folder_more {
  height: 25px;
  margin: 10px;
}

.folder_more:hover {
  filter: invert(0.5)
}

#folder_name {
  margin: 10px;
  font-size: 26px;
}

#folder_name:hover {
  color: grey
}