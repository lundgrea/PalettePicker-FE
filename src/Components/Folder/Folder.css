.Folder {
  display: flex;
}

.folder_delete {
  height: 25px;
}

.folder_more{
  height: 25px;
}