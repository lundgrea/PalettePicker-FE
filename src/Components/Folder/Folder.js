import React from 'react'
import './Folder.css'

export const Folder = (props) => {
  return (
    <article className="Folder">
      <button 
      onClick={(e) => props.getAFoldersPalettes(e, props.id)}
      id="folder_name">{props.name}
      </button>
      <img 
      alt="delete-img"
      id="delete-img" 
      src={require('../../assets/delete.svg')}
      onClick={(e) => props.deleteAFolder(e, props.id)}
      />
    </article>
  )
}

export default Folder
