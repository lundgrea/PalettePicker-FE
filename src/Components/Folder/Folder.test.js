import React from "react";
import { shallow } from "enzyme";
import { Folder } from "./Folder";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let wrapper;
let name = "Blushing"
let palettes = [{name: "palette1", c1: "#FFFFFF", c2: "#FFFFFF", c3: "#FFFFFF", c4: "#FFFFFF", c5: "#FFFFFF", folder_id: 33}, {name: "palette2", c1: "#FFFFFF", c2: "#FFFFFF", c3: "#FFFFFF", c4: "#FFFFFF", c5: "#FFFFFF", folder_id: 33}]

beforeEach(() => {
  wrapper = shallow(<Folder key="1" id="2" name={name} palettes={palettes} getAFoldersPalettes={jest.fn()} deleteAFolder={jest.fn()} onClick={jest.fn()}/>)
});


describe("Folder", () => {
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call getAFoldersPalettes on button click', () => {
    const getAFoldersPalettesMock = jest.fn()
    wrapper = shallow(<Folder key="1" id="2" name={name} palettes={palettes} getAFoldersPalettes={getAFoldersPalettesMock} deleteAFolder={jest.fn()} onClick={jest.fn()}/>)

    const mockEvent = { preventDefault: jest.fn()}
    wrapper.find('button').simulate('click', mockEvent)
    expect(getAFoldersPalettesMock).toHaveBeenCalledWith(mockEvent, "2")
  })

  it('should call deleteAFolder on button click', () => {
    const deleteAFolderMock = jest.fn()
    wrapper = shallow(<Folder key="1" id="2" name={name} palettes={palettes} getAFoldersPalettes={jest.fn()} deleteAFolder={deleteAFolderMock} onClick={jest.fn()}/>)
    const mockEvent = { preventDefault: jest.fn()}
    wrapper.find('img').simulate('click', mockEvent)
    expect(deleteAFolderMock).toHaveBeenCalledWith(mockEvent, "2")
  })
});

