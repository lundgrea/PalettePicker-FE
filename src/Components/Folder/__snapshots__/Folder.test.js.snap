// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder should match snapshot 1`] = `
<article
  className="Folder"
>
  <button
    id="folder_name"
    onClick={[Function]}
  >
    Blushing
  </button>
  <img
    alt="delete-img"
    id="delete-img"
    onClick={[Function]}
    src={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "delete.svg",
      }
    }
  />
</article>
`;
